contracts/USM.sol

```diff
@@ -1,38 +1,66 @@
 pragma solidity ^0.6.7;
 
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
-import "./BufferedToken.sol";
 import "./WadMath.sol";
 import "./FUM.sol";
 import "@nomiclabs/buidler/console.sol";
-
+import "./oracles/IOracle.sol";
 
 /**
  * @title USM Stable Coin
  * @author Alex Roan (@alexroan)
  * @notice Concept by Jacob Eliosoff (@jacob-eliosoff).
  */
-contract USM is BufferedToken {
+contract USM is ERC20 {
     using SafeMath for uint;
     using WadMath for uint;
 
+    address public oracle;
+    FUM public fum;
+    uint public ethPool;                                      // default 0
+    uint public latestFumPrice;                               // default 0
+
+    uint public constant WAD = 10 ** 18;
     uint constant public MIN_ETH_AMOUNT = WAD / 1000;         // 0.001 ETH
     uint constant public MIN_BURN_AMOUNT = WAD;               // 1 USM
     uint constant public MAX_DEBT_RATIO = WAD * 8 / 10;       // 80%
-    uint public latestFumPrice;                               // default 0
-
-    FUM public fum;
 
     event LatestFumPriceChanged(uint previous, uint latest);
 
     /**
      * @param _oracle Address of the oracle
      */
-    constructor(address _oracle) public BufferedToken(_oracle, "Minimal USD", "USM") {
+    constructor(address _oracle) public ERC20("Minimal USD", "USM") {
         fum = new FUM();
+        oracle = _oracle;
         _setLatestFumPrice();
     }
 
+    /**
+     * @notice Calculate the amount of ETH in the buffer
+     *
+     * @return ETH buffer
+     */
+    function ethBuffer() public view returns (int) {
+        int buffer = int(ethPool) - int(_usmToEth(totalSupply()));
+        require(buffer <= int(ethPool), "Underflow error");
+        return buffer;
+    }
+
+    /**
+     * @notice Calculate debt ratio of the current Eth pool amount and outstanding USM
+     * (the amount of USM in total supply).
+     *
+     * @return Debt ratio.
+     */
+    function debtRatio() public view returns (uint) {
+        if (ethPool == 0) {
+            return 0;
+        }
```

Comment:
I'm not sure whether this is needed...  I _think_ once the contract is first funded with a `fund` op (and sliding prices are implemented), the pool is guaranteed to never shrink to 0?  But maybe including this check is safest...

Reply:
Agreed, not needed.